src/queue/multilevel.rs

```diff
@@ -303,6 +293,142 @@ where
+                    .get_metric_with_label_values(&[name])
+                    .unwrap(),
+                array::from_fn(|i| {
+                    TASK_POLL_DURATION
+                        .get_metric_with_label_values(&[name, &format!("{i}")])
+                        .unwrap()
+                }),
+            )
+        } else {
+            (
+                IntCounter::new("_", "_").unwrap(),
+                IntCounter::new("_", "_").unwrap(),
+                Histogram::with_opts(HistogramOpts::new("_", "_")).unwrap(),
+                Histogram::with_opts(HistogramOpts::new("_", "_")).unwrap(),
+                Histogram::with_opts(HistogramOpts::new("_", "_")).unwrap(),
+                array::from_fn(|_| Histogram::with_opts(HistogramOpts::new("_", "_")).unwrap()),
+            )
+        };
+        Self {
+            level0_elapsed_us,
+            total_elapsed_us,
+            task_wait_duration,
+            task_execute_duration,
+            task_execute_times,
+            task_poll_duration,
+        }
+    }
+}
+
+/// TrackedRunnerBuilder is the runner builder for TrackedRunner
+pub struct TrackedRunnerBuilder<B> {
+    inner: B,
+    metrics: MultiLevelMetrics,
+}
+
+impl<B> TrackedRunnerBuilder<B> {
+    pub(super) fn new(inner: B, metrics: MultiLevelMetrics) -> Self {
+        Self { inner, metrics }
+    }
+}
+
+impl<B, R, T> RunnerBuilder for TrackedRunnerBuilder<B>
+where
+    B: RunnerBuilder<Runner = R>,
+    R: Runner<TaskCell = T>,
+    T: TaskCell,
+{
+    type Runner = TrackedRunner<R>;
+
+    fn build(&mut self) -> Self::Runner {
+        TrackedRunner {
+            inner: self.inner.build(),
+            local_level0_elapsed_us: self.metrics.level0_elapsed_us.local(),
+            local_total_elapsed_us: self.metrics.total_elapsed_us.local(),
+            task_execute_duration: self.metrics.task_execute_duration.local(),
+            task_wait_duration: self.metrics.task_wait_duration.local(),
+            task_poll_duration: array::from_fn(|i| self.metrics.task_poll_duration[i].local()),
+            task_execute_times: self.metrics.task_execute_times.local(),
+        }
+    }
+}
+/// The runner for multilevel task queues.
+///
+/// The runner helps multilevel task queues collect additional information.
+/// [`MultilevelRunnerBuilder`] is the [`RunnerBuilder`] for this runner.
+pub struct MultilevelRunner<R> {
+    inner: TrackedRunner<R>,
+    manager: Arc<LevelManager>,
+}
+
+impl<R, T> Runner for MultilevelRunner<R>
+where
+    R: Runner<TaskCell = T>,
+    T: TaskCell,
+{
+    type TaskCell = T;
+
+    fn start(&mut self, local: &mut Local<T>) {
+        self.inner.start(local)
+    }
+
+    fn handle(&mut self, local: &mut Local<T>, task_cell: T) -> bool {
+        let res = self.inner.handle(local, task_cell);
+        let local_total = self.inner.local_total_elapsed_us.get();
+        if local_total > FLUSH_LOCAL_THRESHOLD_US {
             self.manager.maybe_adjust_chance();
```

Comment:
This will be unreachable because `local_total_elapsed_us` is flushed.

Reply:
fixed, PTAL again